hyundai_kia_connect_api/KiaUvoApiEU.py

```diff
@@ -1653,58 +1843,139 @@ def _get_authorization_code_with_form(self, username, password, cookies) -> str:
+
+            response = requests.post(url, headers=headers, data=payload)
+            response_json = response.json()
+            _LOGGER.debug(
+                f"{DOMAIN} - Genesis Get Access Token Response: {response_json}"
+            )
+
+            if "access_token" not in response_json:
+                _LOGGER.error(
+                    f"{DOMAIN} - Genesis access token request failed: {response_json}"
+                )
+                raise AuthenticationError("Failed to get Genesis access token")
+
+            token_type = response_json.get("token_type", "Bearer")
+            access_token = f"{token_type} {response_json['access_token']}"
+            refresh_token = response_json.get("refresh_token", authorization_code)
+
+            _LOGGER.debug(f"{DOMAIN} - Genesis Access Token obtained successfully")
+            return token_type, access_token, refresh_token
+        else:
+            # Original flow for Kia and Hyundai
+            url = self.USER_API_URL + "oauth2/token"
+            headers = {
+                "Authorization": self.BASIC_AUTHORIZATION,
+                "Stamp": stamp,
+                "Content-type": "application/x-www-form-urlencoded",
+                "Host": self.BASE_URL,
+                "Connection": "close",
+                "Accept-Encoding": "gzip, deflate",
+                "User-Agent": USER_AGENT_OK_HTTP,
+            }
+
+            data = (
+                "grant_type=authorization_code&redirect_uri=https%3A%2F%2F"
+                + self.BASE_DOMAIN
+                + "%3A8080%2Fapi%2Fv1%2Fuser%2Foauth2%2Fredirect&code="
+                + authorization_code
+            )
+            _LOGGER.debug(f"{DOMAIN} - Get Access Token Data: {headers}{data}")
+            response = requests.post(url, data=data, headers=headers)
+            response = response.json()
+            _LOGGER.debug(f"{DOMAIN} - Get Access Token Response: {response}")
+
+            token_type = response["token_type"]
+            access_token = token_type + " " + response["access_token"]
+            authorization_code = response["refresh_token"]
+            _LOGGER.debug(f"{DOMAIN} - Access Token Value {access_token}")
+            return token_type, access_token, authorization_code
 
     def _get_refresh_token(self, stamp, authorization_code):
         # Get Refresh Token #
-        url = self.USER_API_URL + "oauth2/token"
-        headers = {
-            "Authorization": self.BASIC_AUTHORIZATION,
-            "Stamp": stamp,
-            "Content-type": "application/x-www-form-urlencoded",
-            "Host": self.BASE_URL,
-            "Connection": "close",
-            "Accept-Encoding": "gzip, deflate",
-            "User-Agent": USER_AGENT_OK_HTTP,
-        }
+        if BRANDS[self.brand] == BRAND_GENESIS:
+            # Genesis specific refresh token flow
+            url = self.USER_API_URL + "oauth2/token"
+
+            payload = {
+                "client_id": self.CCSP_SERVICE_ID,
```

Comment:
For this flow is this the only key difference? If so I would think can reuse code with less duplication.